ftp/src/main/scala/akka/stream/alpakka/ftp/javadsl/FtpApi.scala

```diff
@@ -238,4 +238,16 @@ sealed trait FtpApi[FtpClient] { _: FtpSourceFactory[FtpClient] =>
 
 object Ftp extends FtpApi[FTPClient] with FtpSourceParams
 object Ftps extends FtpApi[FTPClient] with FtpsSourceParams
-object Sftp extends FtpApi[SSHClient] with SftpSourceParams
+object Sftp extends FtpApi[SSHClient] with SftpSourceParams {
+
+  /**
+   * Java API: creates a [[akka.stream.alpakka.ftp.javadsl.FtpApi]] of [[net.schmizz.sshj.SSHClient]] with [[akka.stream.alpakka.ftp.impl.SftpSourceParams]]
+   *
+   * @param customSshClient custom ssh client
+   * @return A [[akka.stream.alpakka.ftp.javadsl.FtpApi]] of [[net.schmizz.sshj.SSHClient]] with [[akka.stream.alpakka.ftp.impl.SftpSourceParams]]
+   */
+  def withSshClient(customSshClient: SSHClient): FtpApi[SSHClient] with SftpSourceParams =
```

Comment:
Would it help the Java API to create a class `Sftp` and base the object on it?
I'd still like to call it `create` or `createWithSshClient` as a dual to Scala's `apply`.

Reply:
Could you make it return a `Sftp` instance instead of `FtpApi[SSHClient]`?

Reply:
In 013f9c9baae32bafbe2f8dbcbc52adb58e81fff3 we tried to create an `SFTP` class that the current `SFTP` object inherited from. This way all existing code using the `SFTP` object would still work.

Sadly Java doesn't play well with this, it is not able to resolve the methods inherited from the traits on companion object as static methods.

We could make a separate type for instead but it would look a bit awkward.

Reply:
What about introducing 
`class SftpApi extends FtpApi[SSHClient] with SftpSourceParams`
and extending and returning that?